src/main/java/org/example/auctions/services/AuctionService.java

```diff
@@ -0,0 +1,93 @@
+package org.example.auctions.services;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+public class AuctionService {
+    private final Map<String, Auction> auctionsCache;
+
+    public AuctionService() {
+        this.auctionsCache = new HashMap<>();
+    }
+
+    public Auction createAuction(String subject, type type, Integer price, duration duration, String ownerName) {
+        if (price <= 0) {
+            throw new RuntimeException("Price is less than zero.");
+        }
+        Auction auction = new Auction(subject, type, price, duration, ownerName);
+        this.auctionsCache.put(auction.getId(), auction);
+        return auction;
+    }
+
+    public void publishAuction(String id) {
```

Comment:
Can we publish an auction which was already published?
Can we publish an auction which was deleted?
Can we publish an auction which was completed?